Check opts of cloud config file

FengyunPan opened the pull request:
Fix #48347
Check opts when register OpenStack CloudProvider rather than
returning error when use opts to create/use cloud resource.

**Release note**:
```release-note
NONE
```

spiffxp commented:
/ok-to-test

FengyunPan commented:
/test pull-kubernetes-e2e-gce-etcd3
ping @anguslees    PTAL
ping @NickrenREN   PTAL

dims commented:
/assign

dims commented:
/approve
/lgtm

dims commented:
/lgtm

dims commented:
/approve

anguslees commented on pkg/cloudprovider/providers/openstack/openstack.go:
```diff
@@ -216,6 +221,40 @@ func readInstanceID() (string, error) {
+
+	// if need to create health monitor for Neutron LB,
+	// monitor-delay, monitor-timeout and monitor-max-retries should be set.
+	emptyDuration := new(MyDuration)
```
No need to create a reference to a `MyDuration` only to deref it again later.  Just do: `emptyDuration := MyDuration{}` here and remove the asterisks below.

FengyunPan replied:
Done.

anguslees commented on pkg/cloudprovider/providers/openstack/openstack_test.go:
```diff
@@ -144,6 +144,98 @@ func TestToAuthOptions(t *testing.T) {
+	delay := new(MyDuration)
+	delay.Duration = 60 * time.Second
+	timeout := new(MyDuration)
+	timeout.Duration = 30 * time.Second
```
Similar to earlier comment - these MyDuration objects should just be regular objects, not references.
```go
delay := MyDuration{60 * time.Second}
timeout := MyDuration{30 * time.Second}
```

FengyunPan replied:
Done.

anguslees commented on pkg/cloudprovider/providers/openstack/openstack.go:
```diff
@@ -53,6 +53,11 @@ const ProviderName = "openstack"
 var ErrNotFound = errors.New("Failed to find object")
 var ErrMultipleResults = errors.New("Multiple results where only one expected")
 var ErrNoAddressFound = errors.New("No address found for host")
+var ErrNoSubnetId = errors.New("subnet-id not set in cloud provider config")
```
It's highly unlikely any caller is going to want to match against one of these different errors explicitly.
So instead of growing a huge list of pre-built errors here, delete these and just use in-line error strings in the function below:

```go
if len(lbOpts.SubnetId) == 0 {
      return fmt.Errorf("subnet-id not set in cloud provider config")
}
// etc
```

FengyunPan replied:
Done. Thank you very much.

anguslees reviewed (changes requested):
some minor style comments

Commit:
Check opts of cloud config file

Fix #48347
Check opts when register OpenStack CloudProvider rather than
returning error when use opts to create/use cloud resource.

dims commented:
/test pull-kubernetes-federation-e2e-gce

dims commented:
/test pull-kubernetes-kubemark-e2e-gce

FengyunPan commented:
／retest

dims commented:
/retest

dims commented:
/lgtm
